ibm/mas_devops/roles/bas_install/tasks/main.yml

```diff
@@ -132,27 +132,88 @@
   no_log: "{{ bas_grafana_password is defined }}"
 
 
-# 7. Create BAS Full Deployment
+# 7. Create BAS AnalyticsProxy
 # -----------------------------------------------------------------------------
 - name: "Create BAS AnalyticsProxy"
   community.kubernetes.k8s:
     definition: "{{ lookup('template', 'templates/analyticsproxy.yaml') }}"
 
 
-# 8. Wait for AnalyticsProxy to be complete
+# 8. BAS Install Reliability Hack
+# -----------------------------------------------------------------------------
+# BAS is VERY unreliable
+# Instead of waiting for the AnalyticsProxy status to report success as we
+# normally would, we need to do some hacky steps to try to handle known problems
+# with the BAS operator that the team are not fixing.
+#
+# If, after 20 minutes, the createcluster job still exists, then we will delete
+# it causing the BAS operator to try again.
+#
+# Note that when the job has successfully completed, the BAS operator deletes
+# the job entirely, so there is no visibility of this job on a good path install.
+# If the job merely exists at this stage then there is a problem with the
+# install, so we don't need to check the state of the job, only that it exists.
+#
+# The BAS operator will now recreate the job, and usually it works on the second
+# attempt.
+#
+# We now resume the normal deployment flow, and will monitor the AnalyticsProxy
+# until it's status.phase is set to "Ready"
+
+- name: "Check the initial status of the AnalyticsProxy"
+  community.kubernetes.k8s_info:
+    api_version: bas.ibm.com/v1
+    name: analyticsproxy
+    namespace: "{{ bas_namespace }}"
+    kind: AnalyticsProxy
+  register: bas_cr_lookup
+  until:
+    - bas_cr_lookup.resources is defined and bas_cr_lookup.resources | length == 1
+    - bas_cr_lookup.resources[0].status is defined
+    - bas_cr_lookup.resources[0].status.phase is defined
+  retries: 15 # approx 15 minutes before we give up waiting for status.phase to be initialized
+  delay: 60 # 1 minute
+
+- name: "Pause for 20 minutes to wait for the createcluster job to complete"
```

Comment:
Any reason to not use a conditional waiting as we do in other places here?
Example: https://github.com/ibm-mas/ansible-devops/blob/master/ibm/mas_devops/roles/amqstreams/tasks/main.yaml#L72

Reply:
I couldn't think of a good way to make it work reliably given the odd way the job is managed by the BAS operator.  Iit not being there could mean either the job hasn't ran at all yet, or it has ran and completed successfully.  This seemed the safest/most reliable way to handle things.  

If the AnalyticsProxy isn't already in Ready state when we start then we wait 15 minutes, and if after 15 minutes this job exists we delete it.  After that we do the normal "wait until Ready" for the AP.

If the AnalyticsProxy is in Ready state already then none of this new code exexcutes (other than the initial check to decide whether to activate this hack).